### PR TITLE
refactor: [M3-7532] - Payment Method Row Storybook v7 Story

### PR DESCRIPTION
**Pull request by coliu-akamai**
## Description 📝
- MIgrate PaymentMethodRow component story to a V7 (.tsx) story

## Changes  🔄
- migrate to .tsx story
- add tests for Delete payment method confirmation dialog
- update imports/exports, switch to tss-mui

### Verification steps 
- Confirm story for Payment Method Row loads as expected: http://localhost:6006/?path=/docs/features-payment-method-row--documentation
- Confirm no regression in appearance on cloud manager
- `yarn test PaymentMethodRow
`
## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/PaymentMethodRow/PaymentMethodRow.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import { paymentMethodFactory } from 'src/factories';
+
+import { PaymentMethodRow } from './PaymentMethodRow';
+
+import type { Meta, StoryObj } from '@storybook/react';
+
+type Story = StoryObj<typeof PaymentMethodRow>;
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const onDelete = () => {};
```
Comment (bnussman-akamai) [inline packages/manager/src/components/PaymentMethodRow/PaymentMethodRow.stories.tsx:13]: Should we use a Storybook action here? 


```suggestion
const onDelete = action('onDelete');
```

File: packages/manager/src/components/PaymentMethodRow/PaymentMethodRow.stories.tsx
Comment (bnussman-akamai) [inline packages/manager/src/components/PaymentMethodRow/PaymentMethodRow.stories.tsx:1]: Could we add a story for an invalid credit card type?

File: packages/manager/src/components/PaymentMethodRow/PaymentMethodRow.tsx [hunk 2]
```diff
@@ -13,14 +13,24 @@ import { Paper } from 'src/components/Paper';
 import CreditCard from 'src/features/Billing/BillingPanels/BillingSummary/PaymentDrawer/CreditCard';
 import { queryKey } from 'src/queries/accountPayment';
 
-import ThirdPartyPayment from './ThirdPartyPayment';
+import { ThirdPartyPayment } from './ThirdPartyPayment';
 
 interface Props {
+  /**
+   * Function called when a payment method is deleted.
```
Comment (bnussman-akamai) [inline packages/manager/src/components/PaymentMethodRow/PaymentMethodRow.tsx:20]:
```suggestion
   * Function called when the delete button in the Action Menu is pressed
```

File: packages/manager/src/components/PaymentMethodRow/DeletePaymentMethodDialog.test.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    expect(deleteButton).toBeVisible();
+    fireEvent.click(deleteButton);
+    expect(props.onDelete).toHaveBeenCalled();
+
+    const cancelButton = screen.getByText('Cancel');
+    expect(cancelButton).toBeVisible();
+    fireEvent.click(cancelButton);
+    expect(props.onClose).toHaveBeenCalled();
+  });
+});
```
Comment (abailly-akamai) [inline packages/manager/src/components/PaymentMethodRow/DeletePaymentMethodDialog.test.tsx:41]: Nice!

nitpick: add a bit of spacing inside and between your tests to make things a bit more readable.

File: packages/manager/src/components/PaymentMethodRow/PaymentMethodRow.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    onDelete,
+    paymentMethod: paymentMethodFactory.build({
+      data: {
+        card_type: 'Visa',
+      },
+      type: 'google_pay',
+    }),
+  },
+  component: PaymentMethodRow,
+  title: 'Features/Payment Method Row',
```
Comment (abailly-akamai) [inline packages/manager/src/components/PaymentMethodRow/PaymentMethodRow.stories.tsx:75]: We should be consistent with the naming/paths of those files. 
If `PaymentMethodRow` is located in `src/components` then it should probably be reflected in the storybook path and not be defined in `features`. (even though `PaymentInformation` is a feature)

**Review by abailly-akamai — APPROVED** [review]
Nice cleanup! 

- story renders well ✅ 
- test look good and passing ✅

**Review by bnussman-akamai — APPROVED** [review]
💳🎉

**Pull request merged**